extensions/amp-ad/0.1/intersection-observer-host.js

```diff
@@ -120,7 +121,7 @@ export function getIntersectionChangeEntry(element, owner, viewport) {
  * over to the iframe if it had not requested the intersection data already via
  * a postMessage.
  */
-export class IntersectionObserver {
+export class IntersectionObserverHost {
```

Comment:
Thank you for moving the file! It has always been very confusing with the name IntersectionObserver

What about naming it `IntersectionObserverHostForAd` or something similar, so we know that it's for ad specifically. (This one will send entry when the element position change even if it is 100% visible).